Services/UI/OJS.Services.Ui.Business/Implementations/ContestsBusinessService.cs

```diff
@@ -112,7 +116,14 @@ public async Task ValidateContestPassword(int id, bool official, string password
 
             var user = this.userProviderService.GetCurrentUser();
 
-            await this.ValidateContest(contest, user.Id, user.IsAdmin, model.IsOfficial);
+            ValidationModel validationModel = new();
+
+            var validationResult = await this.contestValidationService.GetValidationResult((contest, user.Id, user.IsAdmin, model.IsOfficial));
+            if (!validationResult.IsValid)
+            {
+                var propertyInfo = validationModel.GetType().GetProperty(validationResult.PropertyName);
+                propertyInfo?.SetValue(validationModel, false);
```

Comment:
This is great, but move it a method. It is strange to have random reflection inside the business logic :D

Reply:
Also, a comment here will be great.

Reply:
Done.